operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java

```diff
@@ -0,0 +1,256 @@
+import oracle.kubernetes.operator.helpers.KubernetesVersion;
+import oracle.kubernetes.operator.helpers.TuningParametersStub;
+import oracle.kubernetes.operator.work.Step;
+import oracle.kubernetes.operator.work.TerminalStep;
+import oracle.kubernetes.utils.TestUtils;
+import org.hamcrest.Description;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static oracle.kubernetes.operator.StartupControlTest.StepPresenceMatcher.containsStep;
+import static oracle.kubernetes.operator.logging.MessageKeys.CRD_NO_READ_ACCESS;
+import static oracle.kubernetes.operator.logging.MessageKeys.NS_NO_READ_ACCESS;
+import static oracle.kubernetes.utils.LogMatcher.containsWarning;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+public class StartupControlTest {
+  private static final KubernetesVersion VERSION = new KubernetesVersion(1, 8);
+
+  private KubernetesTestSupport testSupport = new KubernetesTestSupport();
+  private Map<String,String> environmentVariables = new HashMap<>();
+  private Function<String, String> getEnv = (name -> environmentVariables.get(name));
+  private List<Memento> mementos = new ArrayList<>();
+  private List<V1ResourceRule> rules = new ArrayList<>();
+  private Step startStep = new TerminalStep();
+  private Step namespaceStep = new TerminalStep();
+  private StartupControl startupControl = new StartupControl(VERSION);
+  private Collection<LogRecord> logRecords = new ArrayList<>();
+  private TestUtils.ConsoleHandlerMemento loggerMemento;
+
+  @Before
+  public void setUp() throws Exception {
+    mementos.add(StaticStubSupport.install(StartupControl.class, "getHelmVariable", getEnv));
+    mementos.add(TuningParametersStub.install());
+    mementos.add(testSupport.install());
+    mementos.add(loggerMemento = TestUtils.silenceOperatorLogger());
+
+    loggerMemento.collectLogMessages(logRecords, CRD_NO_READ_ACCESS, NS_NO_READ_ACCESS);
+    testSupport.doOnCreate(KubernetesTestSupport.SELF_SUBJECT_RULES_REVIEW,
+          (Consumer<V1SelfSubjectRulesReview>) this::populateSubjectRulesReview);
+  }
+
+  @After
+  public void tearDown() {
+    mementos.forEach(Memento::revert);
+  }
+
+  @Test
+  public void whenOperatorDedicatedUndefined_isDedicatedReturnsFalse() {
+    assertThat(startupControl.isDedicated(), is(false));
+  }
+
+  @Test
+  public void whenOperatorDedicatedEmptyString_isDedicatedReturnsFalse() {
+    environmentVariables.put(StartupControl.OPERATOR_DEDICATED_ENV, "");
+    
+    assertThat(startupControl.isDedicated(), is(false));
+  }
+
+  @Test
+  public void whenOperatorDedicatedNotTrue_isDedicatedReturnsFalse() {
+    environmentVariables.put(StartupControl.OPERATOR_DEDICATED_ENV, "zork");
+
```

Comment:
why all the blank lines?

Reply:
Do you mean inside the test methods? That's a common convention to separate individual test setup, the invocation being tested, and the assertion of the expected result.

Reply:
ok